File: server/server-google-cloud-poc/src/main/java/com/googlecloudpoc/service/ItemService.java
package com.googlecloudpoc.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.googlecloudpoc.dto.ItemDto;
import com.googlecloudpoc.dto.ItemDtoMapper;
import com.googlecloudpoc.entity.Item;
import com.googlecloudpoc.repository.ItemRepository;

@Service
public class ItemService {

	private final ItemRepository itemRepository;

	@Autowired
	public ItemService(ItemRepository itemRepository) {
		this.itemRepository = itemRepository;
	}

	public List<ItemDto> getAllItems() {		
		return itemRepository.findAll()
				.stream()
				.map(item -> ItemDtoMapper.toItemDto(item))
				.toList();		
	}

	public ItemDto addItem(ItemDto itemDto) {
		Item savedItem = itemRepository.save(
				ItemDtoMapper.toItem(itemDto)
		);
		return ItemDtoMapper.toItemDto(savedItem);
	}
	
	public void deleteItem(int id) {
		itemRepository.deleteById(id);
	}

}

File: client/client-google-cloud-poc/src/app/service/item.service.ts
import { Injectable } from '@angular/core';
import { ItemRepository } from '../repository/item.repository';
import { ItemDto } from '../dto/item.dto';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ItemService {  

  constructor(private repository: ItemRepository) { }

  getAllItems(): Observable<ItemDto[]> {
    return this.repository.getAllItems();
  }

  addItem(item: ItemDto): Observable<ItemDto> {
    return this.repository.addItem(item);
  }

  deleteItem(id: number): Observable<any> {
    return this.repository.deleteItem(id);
  }
 
}